src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -400,6 +399,109 @@ nlohmann::json SysInfo::getNetworks() const
     return networks;
 }
 
+std::string parseProcessInfo(const std::filesystem::path& path)
+{
+    // Get stat file content.
+    std::string processInfo { UNKNOWN_VALUE };
+    const auto statContent {Utils::getFileContent(path)};
+
+    const auto openParenthesisPos {statContent.find("(")};
+    const auto closeParenthesisPos {statContent.find(")")};
+
+    if (openParenthesisPos != std::string::npos && closeParenthesisPos != std::string::npos)
+    {
+        processInfo = statContent.substr(openParenthesisPos + 1, closeParenthesisPos - openParenthesisPos - 1);
+    }
+
+    return processInfo;
+}
+
+void findInodeMatch(const std::filesystem::path& path, std::vector<int64_t>& inodes, bool& matchInode, const nlohmann::json& portsInfo)
+{
+    const auto fdFileContent {std::filesystem::read_symlink(path)};
+
+    if (!fdFileContent.empty())
+    {
+        // Parse inode from symbolic link. The expected format is: socket[<inode_number>].
+        const auto openBracketPos {fdFileContent.string().find("[")};
+        const auto closeBracketPos {fdFileContent.string().find("]")};
+        const auto match {fdFileContent.string().substr(openBracketPos + 1, closeBracketPos - openBracketPos - 1)};
+        const int64_t matchN {std::stoi(match)};
+
+        if (std::any_of(portsInfo.cbegin(), portsInfo.cend(), [&](const auto it)
+    {
+        return it.at("inode") == matchN;
+        }))
+        {
+            matchInode = true;
+            inodes.push_back(matchN);
+        }
+    }
+}
+
+void parseProcFS(nlohmann::json& portsInfo)
+{
+    const auto Procpath {WM_SYS_PROC_DIR};
+
+    if (std::filesystem::is_directory(Procpath))
```

Comment:
If any throw occurs during those iterations? all data are discarded and the exception is caught by syscollector.

Iterate over the /proc path and create an unordered_map<inode, pair<pid,filepath>>, controlling all handled exceptions.

After this iterate and populate the JSON, handling exceptions inner each iteration.

The unordered_map - json table should be one to one.

Reply:
Reimplemented